### DISCUSSION TITLE
Use of TimeRange for Time dimensions.

### DISCUSSION THREAD

**Discussion by lmorandini** [amattioc/SDMX#215] [General]
Some of the cubes in the new ABS ReST 2.1 endpoints have the dimension TIME_PERIOD with no codes, but an XML TImeRange element instead.

@amattioc : any idea on how to tackle this?

**Comment by amattioc** [comment]
Hi Luca, does the use of Time Range create problems to the connectors? It seems to work fine for me:

![immagine](https://user-images.githubusercontent.com/8512242/75234907-b37c7380-57bb-11ea-9a66-765de2279379.png)

**Comment by lmorandini** [comment]
Could you please point me to a Java code fragment that retrieves the time period (as separate values, such as a list of years) of a cube?

**Comment by amattioc** [comment]
This is the [section](https://github.com/amattioc/SDMX/blob/69e4c4af86b30addb2b57ae1acbf66656cca5c68/JAVA/src/main/java/it/bancaditalia/oss/sdmx/parser/v21/DataStructureParser.java#L346) of the structure parser that handles the TimeDimension. 

The codelists are processed in this [section](https://github.com/amattioc/SDMX/blob/69e4c4af86b30addb2b57ae1acbf66656cca5c68/JAVA/src/main/java/it/bancaditalia/oss/sdmx/parser/v21/DataStructureParser.java#L574) but, since the time dimension is not coded, there is not match for it.

The dsd parsing is successful for the new ABS endpoint, but maybe you are referring to the constraints section?

**Comment by amattioc** [comment]
Hi @lmorandini, any news on this? I was looking at the dataflow message of the new ABS provider and I think that you were probably referring to the constraints section that it contains. Though not specifically needed for the connectors as they are today, thse information could be useful to improve the user experience in building valid queries with the sdmx helper. Unfortunately not many providers send these information yet...

**Comment by lmorandini** [comment]
My issue is more mundane: how to extract the allowed time periods from the GenericSDMXClient/Dataflow/DataFlowStructure classes? I can easily get the code list with getCodeList from DataFlowStructure, but not the time periods: any ideas?

**Comment by amattioc** [comment]
Currently you can not get the time period constraint because it is in a section of the dataflow message that we don't process (because dataflow constraints are not tipically filled by the providers). Nevertheless I think it could be useful to start processing it, when present, for the reason that I explained. I'll keep this issue open, tagging it as an enhancement, and I'll notify you when it is fixed...
